examples/model_service_template/mxnet_vision_batching.py

```diff
@@ -0,0 +1,219 @@
+        self.mx_model.set_params(arg_params, aux_params, allow_missing=True, allow_extra=True)
+
+    def inference(self, model_input):
+        """
+        Internal inference methods for MXNet. Run forward computation and
+        return output.
+
+        :param model_input: list of NDArray
+            Preprocessed inputs in NDArray format.
+        :return: list of NDArray
+            Inference output.
+        """
+        Batch = namedtuple('Batch', ['data'])
+        if self.error is not None:
+            return None
+
+        self.mx_model.forward(Batch([model_input]), is_train=False)
+        outputs = self.mx_model.get_outputs()
+        res = mx.ndarray.split(outputs[0], axis=0, num_outputs=outputs[0].shape[0])
+        res = [res] if not isinstance(res, list) else res
+        return res
+
+    def preprocess(self, request):
+        """
+        Decode all input images into ndarray.
+
+        Note: This implementation doesn't properly handle error cases in batch mode,
+        If one of the input images is corrupted, all requests in the batch will fail.
+
+        :param request:
+        :return:
+        """
+        img_list = []
+        param_name = self.signature['inputs'][0]['data_name']
+        input_shape = self.signature['inputs'][0]['data_shape']
+        # We are assuming input shape is NCHW
+        [c, h, w] = input_shape[1:]
+
+        for idx, data in enumerate(request):
+            img = data.get(param_name)
+            if img is None:
+                img = data.get("body")
+
+            if img is None:
+                img = data.get("data")
+
+            if img is None or len(img) == 0:
+                logging.error("Error processing request")
+                self.erroneous_reqs.add(idx)
+                continue
+
+            try:
+                img_arr = mx.image.imdecode(img, 1, True, None)
+            except Exception as e:
+                logging.warn(e, exc_info=True)
+                self.erroneous_reqs.add(idx)
+                continue
+
+            img_arr = mx.image.imresize(img_arr, w, h, 2)
+            img_arr = mx.nd.transpose(img_arr, (2, 0, 1))
+            self._num_requests = idx + 1
+            img_list.append(img_arr)
+        
+        logging.debug("Worker :{} received {} requests".format(os.getpid(), self._num_requests))
+        reqs = mx.nd.stack(*img_list)
+        reqs = reqs.as_in_context(self.mxnet_ctx)
+
+        if (self._batch_size - self._num_requests) != 0:
+            padding = mx.nd.zeros((self._batch_size - self._num_requests, c, h, w), self.mxnet_ctx, 'uint8')
+            reqs = mx.nd.concat(reqs, padding, dim=0)
+
+        return reqs
+
+    def postprocess(self, data):
+        res = []
+        for idx, resp in data[:self._num_requests]:
+            if idx not in self.erroneous_reqs:
+                res.append(self.top_probability(resp, self.labels, top=5))
+            else:
+                res.append("Illegal request")
```

Comment:
Better keep the error message from earlier stage.

Reply:
I am not sure if I understand the suggestion here. Could you please elaborate?